### ISSUE TITLE
learn.lr_find() doesn't use the specified device

### ISSUE THREAD

**Issue by graftedlife** [fastai/fastai#1186]
**Describe the bug**
I have a computer with two GPUs and wanted to run only on GPU:1 instead of GPU:0.  I specified the device to be torch.device('cuda', 1) when I created the data bunch. When I created learner object using get_tabular_learner and ran the learning rate finder, GPU:1 was used during the training process but GPU:0 was additionally used at the end of training.

Things went on smoothly again when I changed the load function in basic_train.py from

self.model.load_state_dict(torch.load(self.path/self.model_dir/f'{name}.pth', map_location=device))

to 

self.model.load_state_dict(torch.load(self.path/self.model_dir/f'{name}.pth', map_location=torch.device('cuda', 1)))

**To Reproduce**
Load a tabular data and create a learner as in the example doc (https://docs.fast.ai/tabular.html), but add device=torch.device('cuda', 1) in TabularDataBunch.from_df.
Then run:
learn.lr_find()

**Expected behavior**
Only the specified GPU should be used throughout the learning rate finding process.

**Comment by bfarzin** [comment]
```
import fastai
fastai.torch_core.defaults.device = torch.device('cuda',1)
defaults.device
```
Will set the default for this session and then allow you to use just that single GPU for training and `LR_find()`
Does this resolve the issue, or after setting this you still find that you use both GPUs?

**Comment by stas00** [comment]
Won't this do the trick?

```
torch.cuda.set_device(device_id)
```

**Comment by stas00** [comment]
No follow ups, closing this

**Comment by marcdumon** [comment]
> Won't this do the trick?
> 
> ```
> torch.cuda.set_device(device_id)
> ```

torch.cuda.set_device(device_id) doesn't do the trick, but torch.cuda.set_device(device_id) does
